clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -1003,6 +1003,40 @@ public void onAssignment(Set<TopicPartition> assignment) {
         sensors.updatePartitionLagAndLeadSensors(assignment);
     }
 
+    /**
+     * Clear the buffered data (completed fetches) for a set of {@link TopicPartition} which are not a part of newly assigned set of {@link TopicPartition}
+     *
+     * @param assignedTopicPartitions  newly assigned {@link TopicPartition}
+     */
+    public void clearBufferedDataForUnassignedTopicPartitions(Collection<TopicPartition> assignedTopicPartitions) {
+        if (assignedTopicPartitions != null) {
+            Iterator<CompletedFetch> itr = completedFetches.iterator();
+            while (itr.hasNext()) {
+                TopicPartition tp = itr.next().partition;
+                if (!assignedTopicPartitions.contains(tp)) {
+                    itr.remove();
+                }
+            }
+        }
+    }
+
+    /**
+     * Clear the buffered data (completed fetches) for a set of Topics which are not a part of newly assigned set of Topics
+     *
+     * @param assignedTopics  newly assigned topics
+     */
+    public void clearBufferedDataForUnassignedTopics(Collection<String> assignedTopics) {
+        Set<TopicPartition> currentTopicPartitions = new HashSet<>();
+        if (assignedTopics != null) {
```

Comment:
I thought it would be better, since its a public API.

Reply:
It is public API for a class that we are using internally. Thus we are able to enforce rule via code review that `assignedTopics` is never null. And in general it is preferred not to call a function with null parameter, since otherwise we will have a lot of null check in the code which will make the code unnecessarily verbose. So it seems simpler not to check whether it is null.
 
Note that a few other public methods does not do null check. For example `Fetcher.getTopicMetadata` does not check whether request is empty.

And for public APIs that are exposed to user, e.g. `Consumer.assign(partitions)`, because we can not ensure via code review that `partitions` is not null, we currently throw `IllegalArgumentException` is the value is null, instead of allowing user to call the method with null value.